crate/memories/src/sqlite_mem.rs

```diff
@@ -92,34 +65,45 @@ impl<Address, Word> SqliteMemory<Address, Word> {
         })
     }
 
-    /// Returns an `SqliteMemory` instance storing data in a table with the given name
-    /// and connecting to the DB using connections from the given pool.
-    ///
-    /// # Arguments
-    ///
-    /// * `pool` - The pool to use for the memory.
-    /// * `table_name` - The name of the table to create.
-    pub async fn connect_with_pool(
-        pool: Pool,
-        table_name: String,
-    ) -> Result<Self, SqliteMemoryError> {
-        let table_name_clone = table_name.clone();
-        pool.conn(move |conn| conn.execute_batch(&create_table_script(&table_name_clone)))
-            .await?;
-        Ok(Self {
+    /// Returns a new memory instance using a pool of connections to an SQLite
+    /// database.
+    pub async fn new_with_pool(pool: Pool, table_name: String) -> Self {
+        Self {
             pool,
             table_name,
             _marker: PhantomData,
-        })
+        }
     }
-}
 
-impl<Address: Send + Sync, Word: Send + Sync> SqliteMemory<Address, Word> {
+    /// Makes sure the correct table exist in the associated database.
+    pub async fn initialize(&self) -> Result<(), SqliteMemoryError> {
+        // The following settings are used to improve performance:
+        // - journal_mode = WAL : WAL journaling is faster than the default DELETE mode.
+        // - synchronous = NORMAL: Reduces disk I/O by only calling fsync() at critical
+        //   moments rather than after every transaction (FULL mode); this does not
+        //   compromise data integrity.
+        let table_name = format!(
+            "PRAGMA synchronous = NORMAL;
+             PRAGMA journal_mode = WAL;
+             CREATE TABLE IF NOT EXISTS {} (
+                 a BLOB PRIMARY KEY,
+                 w BLOB NOT NULL
+             );",
+            self.table_name
+        );
+
+        self.pool
+            .conn(move |conn| conn.execute_batch(&table_name))
+            .await?;
+        Ok(())
```

Comment:
```suggestion
        let initialization_script = format!(
            "PRAGMA synchronous = NORMAL;
             PRAGMA journal_mode = WAL;
             CREATE TABLE IF NOT EXISTS {} (
                 a BLOB PRIMARY KEY,
                 w BLOB NOT NULL
             );",
            self.table_name
        );

        self.pool
            .conn(move |conn| conn.execute_batch(&initialization_script))
            .await?;
        Ok(())
```
or "table_creation_script", or something of the sort